### PR TITLE
Added yamllint disable-file

### PR DESCRIPTION
**Pull request by deepaknavaria-tanx**
Added a provision to disable yamllint checkup for a particular file.

### CODE REVIEW

File: pre_commit_hooks/run_yamllint.py [hunk 1]
```diff
@@ -3,62 +3,122 @@
...
+    If so, skip the file.
+    """
+    try:
+        with open(yaml_file, "r") as file:
+            first_line = file.readline().strip()
+            return first_line == "# yamllint disable-file"
+    except IOError as e:
+        print(f"Error reading {yaml_file}: {e}")
+        return False
+
```
Comment (Harsh-fi) [inline pre_commit_hooks/run_yamllint.py]: sir, this function is not required as `# yamllint disable-file` is already provided by the package and working fine (I have tested)
reference [https://yamllint.readthedocs.io/en/stable/disable_with_comments.html#disabling-all-checks-for-a-file](https://yamllint.readthedocs.io/en/stable/disable_with_comments.html#disabling-all-checks-for-a-file)

File: pre_commit_hooks/run_yamllint.py [hunk 2]
```diff
@@ -3,62 +3,122 @@
...
+    try:
+        with open(yaml_file, "r") as file:
+            for line in file:
+                # Skip lines containing '# yamllint disable-line'
+                if "# yamllint disable-line" not in line:
+                    processed_lines.append(line)
+        return "".join(processed_lines)
+    except IOError as e:
+        print(f"Error reading {yaml_file}: {e}")
+        sys.exit(1)
```
Comment (Harsh-fi) [inline pre_commit_hooks/run_yamllint.py]: Sir, this function is initialised but never called, please check

File: pre_commit_hooks/run_yamllint.py [hunk 3]
```diff
@@ -3,62 +3,122 @@
...
     if not os.path.isfile(YAMLLINT_CONFIG_FILE):
         print("Yamllint configuration file not found")
         sys.exit(1)
 
+
 def lint_yaml_file(yaml_file, yamllint_path):
+    # Check if the file should be skipped due to 'yamllint disable-file'
+    if should_skip_yaml_file(yaml_file):
+        print(f"Skipping {yaml_file} due to '# yamllint disable-file' directive.")
+        return
```
Comment (Harsh-fi) [inline pre_commit_hooks/run_yamllint.py]: here, call the `preprocess_yaml_file()` function

**Review by Harsh-fi — CHANGES_REQUESTED** [review]
please check the comments

File: pre_commit_hooks/run_yamllint.py (hunk 1 above)
Comment (deepaknavaria-tanx) [reply pre_commit_hooks/run_yamllint.py]: okay

File: pre_commit_hooks/run_yamllint.py (hunk 3 above)
Comment (deepaknavaria-tanx) [reply pre_commit_hooks/run_yamllint.py]: this is resolved

File: pre_commit_hooks/run_yamllint.py (hunk 2 above)
Comment (deepaknavaria-tanx) [reply pre_commit_hooks/run_yamllint.py]: this is also resovled

**Review by Harsh-fi — APPROVED** [review]

**Pull request merged**